### PR TITLE
Add test cases for lazy continuation edge cases

### PR DESCRIPTION
**Pull request by karlb**
lists.test:168 and block_quote.test's lazy cases all put a plain word on the lazy line, and every item or quoted line they continue is a paragraph with content. That leaves the edges of the rule untested: a lazy line that opens a block of its own, an item with nothing to continue, a paragraph opened after a blank inside the item, and a quoted line that is not a paragraph.

### CODE REVIEW

File: test/block_quote.test [hunk 1]
```diff
@@ -82,6 +82,19 @@ lazy</p>
 </blockquote>
 ```
 
+A quoted line that opens a block of its own leaves nothing for a
+later line to continue:
+
```
Comment (jgm) [inline test/block_quote.test:87]: This could be clearer. The relevant point here is that the thematic break block is already closed by the time we get to `lazy`. There is no problem with a quoted line opening a block and then having a lazy continuation:

```
> > a
b
```

File: test/lists.test [hunk 2]
```diff
@@ -181,6 +181,65 @@ two
...
+bare has none, so the next unindented line ends the list; an indented
+one is still the item's content:
+
+```
+-
+next
+
+-
+  next
+.
```
Comment (jgm) [inline test/lists.test:212]: I wonder if we should reconsider this case. After all, it violates the usual expectation that you need a blank line between a list and following paragraph.

Comment (karlb) [reply test/lists.test]: Agreed. I dropped this case from the PR and created a spec issue instead: https://github.com/jgm/djot/issues/413

File: test/block_quote.test (hunk 1 above)
Comment (karlb) [reply test/block_quote.test]: Ok, I rephrased the description.
